docs/eventing/experimental-features/transport-encryption.md

```diff
@@ -0,0 +1,62 @@
+# Transport Encryption for Knative Eventing
+
+**Flag name**: `transport-encryption`
+
+**Stage**: Alpha, disabled by default
+
+**Tracking issue**: [#5957](https://github.com/knative/eventing/issues/5957)
+
+## Overview
+
+By default, event delivery within the cluster is unencrypted. This limits the types of events which
+can be transmitted to those of low compliance value (or a relaxed compliance posture)
+or, alternatively, forces administrators to use a service mesh or encrypted CNI to encrypt the
+traffic, which poses many challenges to Knative Eventing adopters.
+
+Knative Brokers and Channels provides HTTPS endpoints to receive events. Given that these
+endpoints typically do not have public DNS names (e.g. svc.cluster.local or the like), these need to
+be signed by a non-public CA (cluster or organization specific CA).
+
+Event producers are be able to connect to HTTPS endpoints with cluster-internal CA certificates.
+
+## Prerequisites
+
+In order to enable the transport encryption feature, you will need to install cert-manager operator
+by
+following [the cert-manager operator installation instructions](https://cert-manager.io/docs/installation/).
+
+## Transport Encryption configuration
+
+The `transport-encryption` feature flag is an enum configuration that configures how Addressables (
+Broker, Channel, Sink) should accept events.
+
+The possible values for `transport-encryption` are:
+
+- `disabled` (this is equivalent to the current behavior)
+    - Addressables may accept events to HTTPS endpoints
+    - Producers may send events to HTTPS endpoints
+- `permissive`
+    - Addressables should accept events on both HTTP and HTTPS endpoints
+    - Addressables should advertise both HTTP and HTTPS endpoints
+    - Producers should prefer sending events to HTTPS endpoints, if available
+- `strict`
+    - Addressables must not accept events to non-HTTPS endpoints
+    - Addressables must only advertise HTTPS endpoints
+
+For example, to enable `strict` transport encryption, the `config-features` ConfigMap will look like
+the following:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config-features
+  namespace: knative-eventing
+data:
+  transport-encryption: "strict"
```

Comment:
I did just the `kubectl apply` from here: https://cert-manager.io/docs/installation/

```
k get pods -A
NAMESPACE            NAME                                         READY   STATUS    RESTARTS   AGE
cert-manager         cert-manager-5f68c9c6dd-tcms7                1/1     Running   0          2m8s
cert-manager         cert-manager-cainjector-57d6fc9f7d-kh5jc     1/1     Running   0          2m8s
cert-manager         cert-manager-webhook-5b7ffbdc98-zjzvz        1/1     Running   0          2m8s
```

set the value to `strict`, and created a broker. Result:

```
default     my-broker         1s    False   NoAddress
```

Reply:
I missed that we need to apply the new Eventing TLS artifact from the release assets

Reply:
I had it installed via `ko apply -f config/tls/issuers`

same (using the upstream 1.11 branch)

Reply:
That's not enough, you need certificates as well

Reply:
Looks like should be listed in the doc

Reply:
FWIW, doing: 

```
➜  eventing git:(1a5f37cdf) k apply -f https://github.com/knative/eventing/releases/download/knative-v1.11.0/eventing-tls-networking.yaml
```
fixed it :wink: